### PR TITLE
feat:[UI - Swap] Create swap approve dialog

### LINKED ISSUE #15442: [UI - Swap] Create swap approve dialog
Design - https://www.figma.com/design/TS0eQX9dAZXqZtELiwKIoK/Swap---Milestone-1?node-id=3517-435661&m=dev

image

If there are new requirements introduced to a shared component, ensure to add the new support to storybook.
If there are new requirements introduced to a shared component, ensure to add the needed qml tests to cover the changes.
### Description

[comment]: # (Describe the feature you would like and include designs and Figma links if available.)
[comment]: # (Describe the use cases the implementation should cover. E.g. New users, existing users, public chat, group chat, etc)
*Summary*: ...

### Acceptance Criteria

[comment]: # (Rules for the future PR to be accepted.)

### PR DESCRIPTION
**Pull request by caybro**
### What does the PR do

- create a new UI component
- add it to StoryBook
- add QML tests
- integration will be done as part of https://github.com/status-im/status-desktop/issues/15443

Fixes #15442

### Affected areas

SwapApproveCapModal

### Screenshot of functionality (including design for comparison)

- [x] I've checked the design and this PR matches it

image

### CODE REVIEW

**Comment by caybro** [comment]
Putting as WIP, QML tests missing; otherwise should be feature complete

**Review by dlipicar — APPROVED** [review]

File: ui/app/AppLayouts/Wallet/popups/swap/SwapApproveCapModal.qml [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+            Layout.fillWidth: true
+            horizontalAlignment: Qt.AlignHCenter
+            wrapMode: Text.WrapAtWordBoundaryOrAnywhere
+            font.pixelSize: Style.current.additionalTextSize
+            text: qsTr("The smart contract specified will be able to spend up to %1 %2 of your current or future balance.").arg(formatBigNumber(root.fromTokenAmount)).arg(root.fromTokenSymbol)
+        }
+    ]
+
+    headerIconComponent: StatusSmartIdenticon {
+        asset.name: Style.png("swap/paraswap") // FIXME svg
```
Comment (alexjba) [inline ui/app/AppLayouts/Wallet/popups/swap/SwapApproveCapModal.qml:72]: Why not downloading the SVG from Figma? 😄

**Review by alexjba — APPROVED** [review]
LGTM!

Minor detail found on the SwapSignModal. Not related to this PR, but could be a good chance to fix it.
The info tag icon has a wrong color on dark theme

Screenshot 2024-07-08 at 23 44 13

File: ui/app/AppLayouts/Wallet/popups/swap/SwapApproveCapModal.qml (hunk 1 above)
Comment (caybro) [reply ui/app/AppLayouts/Wallet/popups/swap/SwapApproveCapModal.qml:72]: Yeah, we might want to as some point, the current PNG we have is just disgustingly pixelated :)

**Comment by caybro** [comment]
> LGTM!
> 
> Minor detail found on the SwapSignModal. Not related to this PR, but could be a good chance to fix it. The info tag icon has a wrong color on dark theme
> 
> Screenshot 2024-07-08 at 23 44 13

:facepalm: yeah... just forgot to commit/push this part, had it already locally

```diff
--- ui/imports/shared/controls/InformationTag.qml
+++ ui/imports/shared/controls/InformationTag.qml
@@ -60,6 +60,7 @@ Control {
             asset.height: visible ? 16 : 0
             asset.bgHeight: visible ? 16 : 0
             asset.bgWidth: visible ? 16 : 0
+            asset.color: Theme.palette.directColor1
             visible: !!asset.name
         }
         StatusBaseText {
```

Will fix

**Pull request merged**